File: commutil/src/main/java/fz/gb/commutil/network/NetworkType.java
/*
 * Copyright (C) Globalegrow E-Commerce Co. , Ltd. 2007-2018.
 * All rights reserved.
 * This software is the confidential and proprietary information
 * of Globalegrow E-Commerce Co. , Ltd. ("Confidential Information").
 * You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement
 * you entered into with Globalegrow.
 */

package fz.gb.commutil.network;

import android.support.annotation.IntDef;
import android.support.annotation.StringDef;

import java.lang.annotation.RetentionPolicy;

/**
 * 手机网络类型
 *
 * @author dingpeihua
 * @version 1.0
 * @date 2017/11/8 09:05
 */
public class NetworkType {
    /**
     * WiFi
     */
    public static final String NETWORK_WIFI = ("WiFi");
    /**
     * 移动4G
     */
    public static final String NETWORK_4G =("4G");
    /**
     * 移动3G
     */
    public static final String NETWORK_3G =("3G");
    /**
     * 移动2G
     */
    public static final String NETWORK_2G = ("2G");
    /**
     * 未知网络类型
     */
    public static final String NETWORK_UNKNOWN = ("Unknown");
    /**
     * 无网络
     */
    public static final String NETWORK_NO =("No network");

    private String desc;

    @StringDef({NETWORK_WIFI,NETWORK_4G,NETWORK_3G,NETWORK_2G,NETWORK_UNKNOWN,NETWORK_NO})
    public @interface Type{

    }
    NetworkType(String desc) {
        this.desc = desc;
    }

    @Override
    public String toString() {
        return desc;
    }
}

File: commutil/src/main/java/com/fz/common/network/NetworkType.kt
package com.fz.common.network

/**
 * 手机网络类型
 *
 * @author dingpeihua
 * @version 1.0
 * @date 2020/11/25 17:17
 */
enum class NetworkType(private val desc: String) {
    /**
     * ETHERNET
     */
    NETWORK_ETHERNET("ETHERNET"),

    /**
     * WiFi
     */
    NETWORK_WIFI("WiFi"),

    /**
     * 移动5G
     */
    NETWORK_5G("5G"),

    /**
     * 移动4G
     */
    NETWORK_4G("4G"),

    /**
     * 移动3G
     */
    NETWORK_3G("3G"),

    /**
     * 移动2G
     */
    NETWORK_2G("2G"),

    /**
     * 未知移动网络类型
     */
    NETWORK_CELLULAR("CELLULAR"),

    /**
     * 未知网络类型
     */
    NETWORK_UNKNOWN("Unknown"),

    /**
     * 无网络
     */
    NETWORK_NO("No network");

    override fun toString(): String {
        return desc
    }
}